scripts/promote-cherry-picks.ts

```diff
@@ -37,8 +45,52 @@ function getAmpVersionToCherrypick(
   return ampVersionToCherrypick.slice(-13);
 }
 
+async function getCherryPickedPRs(
+  ampVersion: string,
+  numberOfCherryPickedCommits: number
+): Promise<string[]> {
+  try {
+    const {data} = await octokit.rest.repos.listCommits({
+      owner: 'ampproject',
+      repo: 'amphtml',
+      sha: ampVersion,
+      per_page: numberOfCherryPickedCommits,
+    });
+    return data
+      .map(({commit}) => {
+        const [firstLine] = commit.message.split('\n');
+        const matches = firstLine?.match(/\(#(?<pullNumber>\d+)\)$/);
+        return matches?.groups?.pullNumber;
+      })
+      .filter(defined);
```

Comment:
Can the map function be changed to not returned undefined instead of filtering?

Reply:
Well the mapping function has to return _something_ even if that something is `undefined`. We'd still have to filter out the items that didn't succeed in extracting the PR number

Reply:
Why not just use filter and get rid of map?